packages/admin/admin/src/rowActions/RowActionsListItem.tsx

```diff
@@ -4,30 +4,38 @@ import * as React from "react";
 
 import { CommonRowActionItemProps } from "./RowActionsItem";
 
-export type RowActionsListItemComponentsProps = React.PropsWithChildren<{
+export type RowActionsListItemComponentsProps<MenuItemComponent extends React.ElementType = "li"> = React.PropsWithChildren<{
     listItemIcon?: Partial<ListItemIconProps>;
     listItemText?: Partial<ListItemTextProps>;
-    menuItem?: Partial<MenuItemProps>;
+    menuItem?: Partial<MenuItemProps<MenuItemComponent> & { component: MenuItemComponent }>;
 }>;
 
-export interface RowActionsListItemProps extends CommonRowActionItemProps {
+export interface RowActionsListItemProps<MenuItemComponent extends React.ElementType = "li"> extends CommonRowActionItemProps {
     textSecondary?: React.ReactNode;
     endIcon?: React.ReactNode;
-    componentsProps?: RowActionsListItemComponentsProps;
+    componentsProps?: RowActionsListItemComponentsProps<MenuItemComponent>;
     children?: React.ReactNode;
 }
 
-export const RowActionsListItem = React.forwardRef<HTMLLIElement, RowActionsListItemProps>(function RowActionsListItem(props, ref) {
+const RowActionsListItemNoRef = <MenuItemComponent extends React.ElementType>(
+    props: RowActionsListItemProps<MenuItemComponent>,
+    ref: React.ForwardedRef<HTMLLIElement>,
```

Comment:
This should be `MenuItemComponent`, right?

Reply:
Don't get it to work with MenuItemComponent. Should a put in `any` instead?

Reply:
Yes